src/field/globalfield.cxx

```diff
@@ -112,7 +108,7 @@ void GlobalField2D::gather(const Field2D &f) {
   if(mype == data_on_proc) {
     // This processor will receive the data
     
-    MPI_Request* req = new MPI_Request[npes]; // Array of receive handles
+    auto* req = new MPI_Request[npes]; // Array of receive handles
```

Comment:
Could this be `unique_ptr`?

Reply:
I _think_ these could even be `std::vector<MPI_Request>`, and then we pass `req.data()` to MPI. Will investigate!